OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs

```diff
@@ -0,0 +1,168 @@
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModInputCombinationElementMenu : ModMenu, IModInputCombinationElementMenu
+    {
+        public event Action<string> OnConfirm;
+        public event Action OnCancel;
+
+        private ModInputCombinationPopup _inputMenu;
+        private PopupMenu _twoButtonPopup;
+        private IModInputHandler _inputHandler;
+        private SingleAxisCommand _cancelCommand;
+        private string _comboName;
+        private IModInputCombinationMenu _combinationMenu;
+        private IModInputCombinationElement _element;
+
+        public ModInputCombinationElementMenu(IModConsole console, IModInputHandler inputHandler) : base(console)
+        {
+            _inputHandler = inputHandler;
+        }
+
+        private GameObject CreateResetButton(Transform buttonsTransform)
+        {
+            var resetButtonObject = GameObject.Instantiate(buttonsTransform.GetChild(0).gameObject);
+            resetButtonObject.name = "UIElement-ButtonReset";
+            resetButtonObject.transform.SetParent(buttonsTransform);
+            resetButtonObject.transform.SetSiblingIndex(1);
+            resetButtonObject.transform.localScale = buttonsTransform.GetChild(0).localScale;
+            return resetButtonObject;
+        }
+
+        private LayoutManager CreateLayoutManager(GameObject inputObject, GameObject resetButtonObject)
+        {
+            var layoutObject = inputObject.transform.GetChild(0).gameObject;
+            var layoutGroupNew = layoutObject.AddComponent<HorizontalLayoutGroup>();
+            layoutGroupNew.childForceExpandWidth = false;
+            layoutGroupNew.childControlWidth = false;
+            return new LayoutManager(layoutGroupNew, MonoBehaviour.FindObjectOfType<UIStyleManager>(),
+                layoutObject.AddComponent<ModUIStyleApplier>(), resetButtonObject.transform.localScale);
+        }
+
+        public void Initialize(PopupInputMenu menu)
+        {
+            if (Menu != null)
+            {
+                return;
+            }
+            var parentCopy = GameObject.Instantiate(menu.transform.parent.gameObject);
+            parentCopy.AddComponent<DontDestroyOnLoad>();
+            _twoButtonPopup = parentCopy.transform.GetChild(0).GetComponent<PopupMenu>();
+            var originalMenu = parentCopy.transform.GetComponentInChildren<PopupInputMenu>(true);//InputField-Popup
+            var menuTransform = originalMenu.GetComponentInChildren<VerticalLayoutGroup>(true).transform;//InputFieldElements
+            var buttonsTransform = menuTransform.GetComponentInChildren<HorizontalLayoutGroup>(true).transform;
+
+            foreach (var button in buttonsTransform.GetComponentsInChildren<Button>())
+            {
+                button.navigation = new Navigation() { mode = Navigation.Mode.None };
+            }
+            Array.ForEach(menuTransform.GetComponentsInChildren<TabbedNavigation>(), navigation => GameObject.Destroy(navigation));
```

Comment:
`menuTransform.GetComponentsInChildren<TabbedNavigation>().ToList().ForEach(navigation => GameObject.Destroy(navigation));`

Reply:
Isn't that longer though? Both performance and code length-wise

Reply:
Mine is easier to read, no? :)

Reply:
IDK, for me both are equally readable :p

Reply:
Alek's is more readable because of less nesting, but for me neither is very good. An actual `foreach` statement would be way more readable than any one-liner 🤷‍♂️

At the very least do this:
```
var tabNavigations = menuTransform.GetComponentsInChildren<TabbedNavigation>();
Array.ForEach(tabNavigations, navigation => GameObject.Destroy(navigation));
```

Reply:
Agree with @Raicuparta but then I'd do:
```
var tabNavigations = menuTransform.GetComponentsInChildren<TabbedNavigation>().ToList();
tabNavigations.ForEach(navigation => GameObject.Destroy(navigation));
```

Reply:
alright, alright